logical/logical.go

```diff
@@ -2,6 +2,32 @@ package logical
 
 import log "github.com/mgutz/logxi/v1"
 
+// BackendType is the type of backend that is being implemented
+type BackendType uint32
+
+// The these are the types of backends that can be derived from
+// logical.Backend
+const (
+	TypeUnknown     BackendType = 0 // This is also the zero-value for BackendType
+	TypeLogical     BackendType = 1
+	TypeCredential  BackendType = 2
+	TypePassthrough BackendType = 3
```

Comment:
You're right that you wouldn't be able to assert on it, but at the same time, `PassthroughBackend` isn't a plugin :-)

Is there a reason the type assertion can't stay in place for now and we can revisit this if and when a plugin needs it?